src/actions/channel.js

```diff
@@ -220,9 +220,7 @@ export const openChannel = (channelId, messageId) => (dispatch, getState) => {
     })
     .then(([channel, users]) => {
       if (channel && users) {
-        users.forEach((user) => {
-          dispatch(addUser(user))
-        })
+        dispatch(addUser(channel))
```

Comment:
passing in the channel into addUser … is this correct?

Reply:
@nikgraf yes, the channel can be two types 'pm' and 'channel'

Reply:
I'm confused. You have a function addUser https://github.com/ubergrape/grape-web-client/blob/21f3f23f84008ef69747120b0b5f42e550b0c2a6/src/actions/common.js#L65 and you passing in a channel. That doesn't make sense :)

What is addUser doing there. It seems to expect a user object not a channel

Reply:
@nikgraf I'm simplified a bit code, and rename channel to `pmChannel`, because it's what we have the same structure for `get_channel` (if the channel is PM) - https://uebergrape.staging.chatgrape.com/doc/chat_api/rpc.html#chat.rpc.Channels.get

And for `get_overview` for PM https://uebergrape.staging.chatgrape.com/doc/chat_api/rpc.html#chat.rpc.Channels.get

`get_overview` for PM we store as `users` in Redux

Is still confusing?

Reply:
Agree with @nikgraf. `addUser(channel)` expects a user object to be passed in.